chirp/drivers/retevis_ha1g.py

```diff
@@ -0,0 +1,1922 @@
+0xB101, 0x71C0, 0x7080, 0xB041, 0x5000, 0x90C1, 0x9181, 0x5140, 0x9301, 0x53C0, 0x5280, 0x9241, 0x9601, 0x56C0, 0x5780, 0x9741,
+0x5500, 0x95C1, 0x9481, 0x5440, 0x9C01, 0x5CC0, 0x5D80, 0x9D41, 0x5F00, 0x9FC1, 0x9E81, 0x5E40, 0x5A00, 0x9AC1, 0x9B81, 0x5B40,
+0x9901, 0x59C0, 0x5880, 0x9841, 0x8801, 0x48C0, 0x4980, 0x8941, 0x4B00, 0x8BC1, 0x8A81, 0x4A40, 0x4E00, 0x8EC1, 0x8F81, 0x4F40,
+0x8D01, 0x4DC0, 0x4C80, 0x8C41, 0x4400, 0x84C1, 0x8581, 0x4540, 0x8701, 0x47C0, 0x4680, 0x8641, 0x8201, 0x42C0, 0x4380, 0x8341,
+0x4100, 0x81C1, 0x8081, 0x4040,
+]
+
+MEM_FORMAT = """
+   
+#seekto 0x0E;
+struct{
+   u8 modelnumber[32];
+   u8 hardwareversion[2];
+   u8 serialn0[16];
+   u8 freqmin[4];
+   u8 freqmax[4];
+   u8 rev_1;
+   u8 saleterr;
+   u8 radiomode;
+   u8 rev[7];
+    } radioinfo;
+
+#seekto 0x52;
+struct{
+    u8 dataver[2];
+    u8 softver[4];
+    u8 rev[4];
+    }dataver;
+
+#seekto 0x5C;
+struct {
+  u8 readpwd[8];          
+  u8 writepwd[8];
+  u8 poweronpwd[8];
+  u8 squelch:4,
+     rev_1:4;           
+  u8 powersavingdelaytime:4,
+     powersavingmode:4;
+  u8 tk1_short;
+  u8 tk1_long;
+  u8 sk1_short;
+  u8 sk1_long;
+  u8 sk2_short;
+  u8 sk2_long;
+  u8 longtime;
+  u8 chknoblock:1,
+     sidekeylock:1,
+     panellock:1,
+     rev_2:3,
+     autoormanuallock:1,
+     keylockswitch:1;
+  u8 wcscan:1,
+     agc:1,
+     rtcreport:1,
+     displaybatterylevel:1,
+     batterlevelreport:1,
+     lowbattery:1,
+     stunmode:2;
+  u8 voiceprompts:1,
+     touchtone:1,
+     txenstone:1,
+     channelmode:1,
+     screenlight:1,
+     rogerbeep:1,
+     rev_3:2;
+  u8 homepoweronzone_1;
+  u8 poweron_type_1:4,
+     homepoweronzone_1_height:4;
+  u16 homepoweronch_1;
+  u8 relaytaildelay;
+  u8 radioid[10];
+  u8 radioname[16];
+  u8 micmain;
+  u8 agcmain:4,
+     agcthreshold:4;
+  u8 scenemode;
+  u8 autooff;
+  u8 menuouttime;
+  u8 voxthreshold:4,
+     voxdelaytime:4;
+  u8 rev_4:1,     
+     homeselect:3,
+     homeindex:2,
+     chdisplay:2;
+  u8 homepoweronzone_2;
+  u8 poweron_type_2:4,
+     homepoweronzone_2_height:4;
+  u16 homepoweronch_2;
+  u16 homepoweronzone_3;
+  u16 homepoweronch_3;
+  u16 frestep;
+  u8 language;
+  u8 backlightbrightness:4,
+     backlighttime:4;
+  u8 wxch;
+  u8 calltone;
+  u8 rev_5:5,    
+     homechtype_3:1,
+     homechtype_2:1,
+     homechtype_1:1;
+  u8 spreadspectrummode;
+  u8 salezone;
+  u8 tailsoundeliminationsfre:7,
+     tailsoundeliminationswitch:1;
+  u8 singletone;
+  u8 scanlist;
+  u8 rev_6[6];
+} settings;
+
+#seekto 0xc0;
+struct  {
+    u16 zonenum;
+    u16 zoneindex[64]; 
+} zonedata; 
+
+#seekto 0x142;
+struct {
+    char name[14];
+    u16 chnum;
+    u16 chindex[16];
+} zones[64];
+
+#seekto 0x0D42;
+struct  {
+    u16 chnum;
+    u16 chindex[1027];
+} channeldata;
+
+#seekto 0x154a;
+struct  {
+  u8 alias[14];  
+  u8 chmode:4,
+     chpro:4;
+  u8 fixedpower:1,
+     fixedbandwidth:1,
+     TailElimination:2,
+     power:2,
+     bandwidth:2;
+  u8 rxfreq[4];
```

Comment:
From the code below, it looks like this should be `ul32 rxfreq`. You should never need `int.frombytes()`. It looks like there are many other cases here that should be proper data types and not just `u8` arrays.